src/ui/suspense/frameCache.ts

```diff
@@ -0,0 +1,18 @@
+import { Frame, PauseId } from "@replayio/protocol";
+import { createGenericCache } from "@bvaughn/src/suspense/createGenericCache";
+import { Pause } from "protocol/thread/pause";
+import { assert } from "protocol/utils";
+
+export const {
+  getValueSuspense: getFramesSuspense,
+  getValueAsync: getFramesAsync,
+  getValueIfCached: getFramesIfCached,
+} = createGenericCache<[PauseId], Frame[] | undefined>(
+  async pauseId => {
+    const pause = Pause.getById(pauseId);
+    assert(pause, `no pause for ${pauseId}`);
+    const wiredFrames = await pause.getFrames();
+    return wiredFrames?.map(f => pause.rawFrames.get(f.frameId)!);
+  },
+  pauseId => pauseId
+);
```

Comment:
I'm curious about these caches. Is there a reason they're in the main parent directory instead of in `@bvaughn/src/suspense` where we could add some test coverage with the new type of tests? Is it to avoid adding `ReplayClientInterface` methods for them or something else?

It's not a problem. I'm just curious.

Reply:
I put them there because they reference `ThreadFront` (for now) and I didn't want to introduce that dependency to `packages/bvaughn-architecture-demo`.

Reply:
Right. We shouldn't ever depend on TF in the new package, but on `shared/client`.